pkg/registry/thirdpartyresourcedata/util.go

```diff
@@ -20,9 +20,28 @@ import (
 	"fmt"
 	"strings"
 
+	"k8s.io/kubernetes/pkg/api/unversioned"
 	"k8s.io/kubernetes/pkg/apis/extensions"
 )
 
+func ExtractGroupVersionKind(list *extensions.ThirdPartyResourceList) ([]unversioned.GroupVersion, []unversioned.GroupVersionKind, error) {
```

Comment:
Yeah, but there are functions that need just `[]GroupVersion` (in multiple places) so its easier to put the extract code and re-array code in one function.